addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2560,7 +2574,11 @@ const RangeUserValueWidget = UnitUserValueWidget.extend({
         const inputValue = possibleValues.length > 1 ? possibleValues.indexOf(value) : this._value;
         this.input.value = inputValue;
         if (this.displayValue) {
-            this.outputEl.value = inputValue;
+            let c = Number(inputValue);
+            if (this.exponential && c >= 0) {
```

Comment:
IMO it's not really `this.exponential` it's rather `positiveNumberExponential` or `exponentialAboveZero` which should be said .. to be clear..